### PR TITLE
Sprint Rewrite

### PR DESCRIPTION
**Pull request by CrytoPal**
## Type of change

- [ ] Bug fix
- [x] New feature

## Description

Adds Modes to Sprint (Strict, Rage, Omni) + adds a check for hunger so it doesn't flag Anticheats

## Related issues

nothing

# How Has This Been Tested?

AC testing servers (Grim, and NCP) + Survival world

# Checklist:

- [x] My code follows the style guidelines of this project.
- [x] I have added comments to my code in more complex areas.
- [x] I have tested the code in both development and production environments.

### CODE REVIEW

**Comment by headzz06** [comment]
why is currentmode a global variable? also a switch case might be a better idea, or at least an else if

File: src/main/java/meteordevelopment/meteorclient/systems/modules/movement/Sprint.java [hunk 1]
```diff
@@ -7,37 +7,82 @@
...
+
+    // Removed whenStationary as it was just Rage sprint
+    private final Setting<Boolean> hunger = sgGeneral.add(new BoolSetting.Builder()
+        .name("check-for-hunger")
+        .description("Checks if you have enough hunger to sprint")
         .defaultValue(false)
         .build()
     );
 
+
```
Comment (arlomcwalter) [inline src/main/java/meteordevelopment/meteorclient/systems/modules/movement/Sprint.java:42]: ?

File: src/main/java/meteordevelopment/meteorclient/systems/modules/movement/Sprint.java [hunk 2]
```diff
@@ -7,37 +7,82 @@
...
     public Sprint() {
         super(Categories.Movement, "sprint", "Automatically sprints.");
     }
 
+    public Mode currentmode;
```
Comment (arlomcwalter) [inline src/main/java/meteordevelopment/meteorclient/systems/modules/movement/Sprint.java:47]: fields go above constructors, shouldn't be public and should be written currentMode

File: src/main/java/meteordevelopment/meteorclient/systems/modules/movement/Sprint.java [hunk 3]
```diff
@@ -7,37 +7,82 @@
...
         mc.player.setSprinting(false);
     }
 
     @EventHandler
     private void onTick(TickEvent.Post event) {
-        if (mc.player.forwardSpeed > 0 && !whenStationary.get()) {
-            mc.player.setSprinting(true);
-        } else if (whenStationary.get()) {
-            mc.player.setSprinting(true);
+        currentmode = mode.get();
```
Comment (arlomcwalter) [inline src/main/java/meteordevelopment/meteorclient/systems/modules/movement/Sprint.java:56]: why is this even a thing, just do switch (mode.get()) { ... }
the variable shouldn't exist

File: src/main/java/meteordevelopment/meteorclient/systems/modules/movement/Sprint.java [hunk 4]
```diff
@@ -7,37 +7,82 @@
...
+        if (currentmode == Mode.Omni) {
```
Comment (arlomcwalter) [inline src/main/java/meteordevelopment/meteorclient/systems/modules/movement/Sprint.java:57]: make this all (the rest of the code) into a switch statement

Comment (CrytoPal) [reply src/main/java/meteordevelopment/meteorclient/systems/modules/movement/Sprint.java]: Alright

**Comment by CrytoPal** [comment]
Alright the issues you stated @arlomcwalter

**Comment by NCPUser** [comment]
stop 2b2t use GrimAC?

**Comment by NCPUser** [comment]
Nice sprint

**Pull request merged**